core/autoscaling_test.go

```diff
@@ -1086,6 +1086,22 @@ func TestNeedReplaceOnDemandInstances(t *testing.T) {
 			desiredCapacity: aws.Int64(0),
 			expectedRun:     false,
 		},
+		{name: "ASG has only one remaining instance, less than enough on-demand",
+			asgInstances: makeInstancesWithCatalog(
+				map[string]*instance{
+					"id-1": {
+						Instance: &ec2.Instance{
+							State:             &ec2.InstanceState{Name: aws.String("running")},
+							Placement:         &ec2.Placement{AvailabilityZone: aws.String("eu-west-1a")},
+							InstanceLifecycle: aws.String("spot"),
+						},
+					},
+				},
+			),
+			minOnDemand:     1,
+			desiredCapacity: aws.Int64(0),
```

Comment:
Shouldn't we also have a case for min=desired=1?

Reply:
Indeed, thanks for check